### PR TITLE
ESQL: Fix attribute set equals

### PR DESCRIPTION
**Pull request by alex-spies**
`AttributeSet.equals` was made so that one set was never ever equal to another; because we used the underlying `AttributeMap` and delegated to `AttributeMap.equals`, which requires equal classes first and foremost.

Fix that, following the pattern from `AttributeMap.equals`.

And, while at it, let's add a test that uses this and checks that name ids from different `LOOKUP JOIN` commands do not overlap.

### CODE REVIEW

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java [hunk 1]
```diff
@@ -2190,6 +2191,36 @@ public void testLookupJoinUnknownField() {
         assertThat(e.getMessage(), containsString(errorMessage3 + "right side of join"));
     }
 
+    public void testMultipleLookupJoinsGiveDifferentAttributes() {
```
Comment (alex-spies) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java:2194]: @craigtaverner , we were chatting about attributes obtained from different `LOOKUP JOINs`. This test demonstrates that they do the right thing!

File: x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/AttributeSet.java [hunk 2]
```diff
@@ -174,8 +174,12 @@ public Stream<Attribute> parallelStream() {
...
 
     @Override
-    public boolean equals(Object o) {
-        return delegate.equals(o);
+    public boolean equals(Object obj) {
+        if (obj instanceof AttributeSet as) {
+            obj = as.delegate;
+        }
+
+        return delegate.equals(obj);
```
Comment (idegtiarenko) [inline x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/AttributeSet.java:182]: Do we expect it to ever be equal if object is not `AttributeSet`? If not, following could avoid reassigning the parameter
```suggestion
        return obj instanceof AttributeSet as && Objects.equals(as.delegate, this.delegate);
```

Comment (alex-spies) [reply x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/AttributeSet.java:182]: Yeah, I think your approach would be stricter and I'd write it like this if I was writing this from scratch.

However, that'd potentially rule previously equal sets to not be equal anymore, even though that probably never applied in reality. But we'd have to go and check. And this'd be inconsistent with `AttributeMap`, which also defers to `delegate.equals` even for instances of different classes.

**Review by costin — APPROVED** [review]
Thanks for catching this. LGTM

**Comment by alex-spies** [comment]
The serverless checks fail for a single, completely unrelated test failure (already tracked in https://github.com/elastic/elasticsearch-serverless/issues/3313), so I believe this is fine to merge regardless.

**Pull request merged**